File: src/service/ModelBuilder.Core/Models/Node.cs
using System;
using System.Collections.Generic;
using Mb.Core.Enums;

namespace Mb.Core.Models
{
    public class Node
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; }
        public IconType Icon { get; set; }
        public string Label { get; set; }
        public NodeType Type { get; set; }
        public decimal PositionX { get; set; }
        public decimal PositionY { get; set; }
        public virtual ICollection<Connector> Connectors { get; set; }
        public bool IsLocked { get; set; }
        public bool IsSelected { get; set; }
        public virtual ICollection<Attribute> Attributes { get; set; }
        public virtual ICollection<Project> Projects { get; set; }
    }
}

File: src/client/src/models/data/Node.ts
import { Purpose } from "../enums/Purpose";
import { EnumBase } from "../enums/EnumBase";
import { Aspect } from "../enums/Aspect";
import { Connector } from "./Connector";
import { Attribute } from "./Attribute";
import { Simple } from "./Simple";

export const NODE_KIND = "Node";

export interface Node {
  id: string;
  iri: string;
  domain: string;
  projectId: string;
  projectIri: string;
  rds: string;
  description: string;
  semanticReference: string;
  tagNumber: string;
  name: string;
  label: string;
  positionX: number;
  positionY: number;
  purpose: Purpose;
  positionBlockX: number;
  positionBlockY: number;
  level: number;
  order: number;
  statusId: string;
  status: EnumBase;
  createdBy: string;
  libraryTypeId: string;
  created: Date;
  updatedBy: string;
  updated: Date;
  version: string;
  aspect: Aspect;
  isRoot: boolean;
  masterProjectId: string;
  masterProjectIri: string;
  symbol: string;
  connectors: Connector[];
  attributes: Attribute[];
  simples: Simple[];
  width: number;
  height: number;
  isLocked: boolean;
  isLockedStatusBy: string;
  isLockedStatusDate: string;
  kind: string;

  // Only for client
  parentNodeId: string;
  selected?: boolean;
  blockSelected?: boolean;
  hidden?: boolean;
  blockHidden?: boolean;
  isOffPageTarget?: boolean;
  isOffPageRequired?: boolean;
}
